src/software/field_tests/field_test_fixture.py

```diff
@@ -0,0 +1,605 @@
+PROCESS_BUFFER_DELAY_S = 0.01
+PAUSE_AFTER_FAIL_DELAY_S = 3
+LAUNCH_DELAY_S = 0.1
+TEST_END_DELAY = 0.3
+
+
+class FieldTestRunner(TbotsTestRunner):
+    """Run a field test"""
+
+    def __init__(
+        self,
+        test_name,
+        thunderscope,
+        blue_full_system_proto_unix_io,
+        yellow_full_system_proto_unix_io,
+        gamecontroller,
+        publish_validation_protos=True,
+    ):
+        """Initialize the FieldTestRunner
+        
+        :param test_name: The name of the test to run
+        :param blue_full_system_proto_unix_io: The blue full system proto unix io to use
+        :param yellow_full_system_proto_unix_io: The yellow full system proto unix io to use
+        :param gamecontroller: The gamecontroller context managed instance 
+        :param publish_validation_protos: whether to publish validation protos
+
+        """
+
+        super(FieldTestRunner, self).__init__(
+            test_name,
+            thunderscope,
+            blue_full_system_proto_unix_io,
+            yellow_full_system_proto_unix_io,
+            gamecontroller,
+        )
+        self.publish_validation_protos = publish_validation_protos
+
+        logger.info("determining robots on field")
+        # survey field for available robot ids
+        try:
+            world = self.world_buffer.get(block=True, timeout=WORLD_BUFFER_TIMEOUT)
+            self.initial_world = world
+            self.friendly_robot_ids_field = [
+                robot.id for robot in world.friendly_team.team_robots
+            ]
+            self.enemy_robot_ids_field = [
+                robot.id for robot in world.enemy_team.team_robots
+            ]
+
+            logger.info(f"blue team ids {self.friendly_robot_ids_field}")
+
+            if len(self.friendly_robot_ids_field) == 0:
+                print("raising an exception", flush=True)
+                raise Exception("no friendly robots found on field")
+
+        except queue.Empty as empty:
+            raise Exception("unable to determine robots on the field")
+
+    def send_gamecontroller_command(
+        self,
+        gc_command: proto.ssl_gc_state_pb2.Command,
+        team: proto.ssl_gc_common_pb2.Team,
+        final_ball_placement_point=None,
+    ):
+
+        self.gamecontroller.send_ci_input(
+            gc_command=gc_command,
+            team=team,
+            final_ball_placement_point=final_ball_placement_point,
+        )
+
+    def set_worldState(self, worldstate: WorldState):
+        """ Sets a world state on the field by moving robots and the ball to the given positions
+
+        Args:
+            worldstate (WorldState): Proto which contains robot and ball positions expected on the field
+
+        Raises:
+            Exception: On failing to verify that the expected position was set
+        """
+
+        self.__validateMatchingRobotIds(worldstate)
+        self.__setBallState(worldstate)
+        self.__setRobotState(worldstate)
```

Comment:
It doesn't actually seem like `set_worldState` is called in the example test. Do we know if this works and the robots can setup the test prior to running? If not and it is outside of the scope of this PR, then please add a TODO for this to be validated/fixed

Reply:
Did you mention that the robots are now able to position themselves? Has that been pushed yet?